### PR TITLE
Document AKSClusterConfig API

### PR DESCRIPTION
**Pull request by mjura**
Issue: https://github.com/rancher/aks-operator/issues/238

### CODE REVIEW

File: docs/resources.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#### Arguments for AKSClusterConfigSpec is the spec for a AKSClusterConfig resource
+
+The following arguments are supported:
+
+* `Imported` - (Optional) Is AKS cluster imported? Defaul: `false` (bool)
```
Comment (richardcase) [inline docs/resources.md:5]: These descriptions can be moved to comments on the `AKSClusterConfigSpec` and with the additional of the kubebuilder validation annotations.

File: pkg/apis/aks.cattle.io/v1/types.go [hunk 2]
```diff
@@ -33,40 +33,112 @@ type AKSClusterConfig struct {
...
-	PrivateCluster              *bool             `json:"privateCluster"`
-	PrivateDNSZone              *string           `json:"privateDnsZone" norman:"pointer"`
-	AuthorizedIPRanges          *[]string         `json:"authorizedIpRanges" norman:"pointer"`
-	HTTPApplicationRouting      *bool             `json:"httpApplicationRouting"`
-	Monitoring                  *bool             `json:"monitoring"`
-	LogAnalyticsWorkspaceGroup  *string           `json:"logAnalyticsWorkspaceGroup" norman:"pointer"`
-	LogAnalyticsWorkspaceName   *string           `json:"logAnalyticsWorkspaceName" norman:"pointer"`
-	ManagedIdentity             *bool             `json:"managedIdentity" norman:"pointer"`
-	UserAssignedIdentity        *string           `json:"userAssignedIdentity" norman:"pointer"`
+	// Importer - Whether to import the cluster or not.
```
[the remark this answers is not in the document]
Comment (mjura) [reply pkg/apis/aks.cattle.io/v1/types.go]: fixed

**Review by mbologna — APPROVED** [review]

File: pkg/apis/aks.cattle.io/v1/types.go [hunk 3]
```diff
@@ -33,40 +37,113 @@ type AKSClusterConfig struct {
...
-	Monitoring                  *bool             `json:"monitoring"`
-	LogAnalyticsWorkspaceGroup  *string           `json:"logAnalyticsWorkspaceGroup" norman:"pointer"`
-	LogAnalyticsWorkspaceName   *string           `json:"logAnalyticsWorkspaceName" norman:"pointer"`
-	ManagedIdentity             *bool             `json:"managedIdentity" norman:"pointer"`
-	UserAssignedIdentity        *string           `json:"userAssignedIdentity" norman:"pointer"`
+	// Importer indicates that the cluster was imported.
+	// +optional
+	// +kubebuilder:default=false
+	Imported bool `json:"imported" norman:"noupdate"`
+	// Location specifies the region to create the private endpoint.
```
Comment (furkatgofurov7) [inline pkg/apis/aks.cattle.io/v1/types.go:44]: ResourceLocation

File: pkg/apis/aks.cattle.io/v1/types.go [hunk 4]
```diff
@@ -33,40 +37,113 @@ type AKSClusterConfig struct {
...
-	PrivateCluster              *bool             `json:"privateCluster"`
-	PrivateDNSZone              *string           `json:"privateDnsZone" norman:"pointer"`
-	AuthorizedIPRanges          *[]string         `json:"authorizedIpRanges" norman:"pointer"`
-	HTTPApplicationRouting      *bool             `json:"httpApplicationRouting"`
-	Monitoring                  *bool             `json:"monitoring"`
-	LogAnalyticsWorkspaceGroup  *string           `json:"logAnalyticsWorkspaceGroup" norman:"pointer"`
-	LogAnalyticsWorkspaceName   *string           `json:"logAnalyticsWorkspaceName" norman:"pointer"`
-	ManagedIdentity             *bool             `json:"managedIdentity" norman:"pointer"`
-	UserAssignedIdentity        *string           `json:"userAssignedIdentity" norman:"pointer"`
+	// Importer indicates that the cluster was imported.
```
Comment (furkatgofurov7) [inline pkg/apis/aks.cattle.io/v1/types.go:40]: Imported*

File: pkg/apis/aks.cattle.io/v1/types.go [hunk 5]
```diff
@@ -33,40 +37,113 @@ type AKSClusterConfig struct {
...
+	// Immutable.
+	// +optional
+	NetworkDNSServiceIP *string `json:"dnsServiceIp" norman:"pointer"`
+	// NetworkService CIDR is the network service cidr.
+	NetworkServiceCIDR *string `json:"serviceCidr" norman:"pointer"`
+	// NetworkDockerBridgeCIDR is the network docker bridge cidr.
+	NetworkDockerBridgeCIDR *string `json:"dockerBridgeCidr" norman:"pointer"`
+	// NetworkPodCIDR is the network pod cidr.
+	NetworkPodCIDR *string `json:"podCidr" norman:"pointer"`
+	// NodeResourceGroupName is the name of the resource group
```
Comment (furkatgofurov7) [inline pkg/apis/aks.cattle.io/v1/types.go:84]: NodeResourceGroup

File: pkg/apis/aks.cattle.io/v1/types.go [hunk 6]
```diff
@@ -33,40 +37,113 @@ type AKSClusterConfig struct {
...
+	// containing cluster IaaS resources.
+	// +optional
+	NodeResourceGroup *string `json:"nodeResourceGroup,omitempty" norman:"pointer"`
+	// Outbound configuration used by Nodes.
```
Comment (furkatgofurov7) [inline pkg/apis/aks.cattle.io/v1/types.go:88]: OutboundType

File: pkg/apis/aks.cattle.io/v1/types.go [hunk 7]
```diff
@@ -33,40 +37,113 @@ type AKSClusterConfig struct {
...
+	NetworkPolicy *string `json:"networkPolicy" norman:"pointer"`
+	// LinuxAdminUsername is a string literal containing a linux admin username.
+	// +optional
+	LinuxAdminUsername *string `json:"linuxAdminUsername,omitempty" norman:"pointer"`
+	// LinuxSSHPublicKey is a string literal containing a ssh public key.
+	// +optional
+	LinuxSSHPublicKey *string `json:"sshPublicKey,omitempty" norman:"pointer"`
+	// DNSPrefix is the DNS prefix to use with hosted Kubernetes API server FQDN.
+	DNSPrefix *string `json:"dnsPrefix,omitempty" norman:"pointer"`
+	// Version defines the desired Kubernetes version.
```
Comment (furkatgofurov7) [inline pkg/apis/aks.cattle.io/v1/types.go:112]: KubernetesVersion

File: pkg/apis/aks.cattle.io/v1/types.go [hunk 8]
```diff
@@ -76,21 +153,51 @@ type AKSClusterConfigStatus struct {
...
-	MaxSurge            *string            `json:"maxSurge,omitempty"`
-	MaxCount            *int32             `json:"maxCount,omitempty"`
-	MinCount            *int32             `json:"minCount,omitempty"`
-	EnableAutoScaling   *bool              `json:"enableAutoScaling,omitempty"`
-	VnetSubnetID        *string            `json:"vnetSubnetID,omitempty" norman:"pointer"`
-	NodeLabels          map[string]*string `json:"nodeLabels,omitempty"`
-	NodeTaints          *[]string          `json:"nodeTaints,omitempty"`
+	// Name is the name of the node pool.
+	Name *string `json:"name,omitempty" norman:"pointer"`
+	// NodeCount is the number of nodes in the node pool.
```
Comment (furkatgofurov7) [inline pkg/apis/aks.cattle.io/v1/types.go:158]: Count

**Comment by cpinjani** [comment]
Can we backport to release-v2.8 please, it would unblock QA validation for https://github.com/rancher/aks-operator/issues/238

**Pull request merged**